src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs

```diff
@@ -83,67 +83,24 @@ public virtual Task InvokeAsync()
                 return Awaited(this, task, scope);
             }
 
-            Exception? releaseException = null;
-            try
-            {
-                ReleaseResources();
-            }
-            catch (Exception exception)
-            {
-                releaseException = exception;
-            }
-
-            Exception? scopeException = null;
-            try
-            {
-                scope?.Dispose();
-            }
-            catch (Exception exception)
-            {
-                scopeException = exception;
-            }
-
-            if (releaseException == null && scopeException == null)
-            {
-                return Task.CompletedTask;
-            }
-            else if (releaseException != null && scopeException != null)
-            {
-                return Task.FromException(new AggregateException(releaseException, scopeException));
-            }
-            else if (releaseException != null)
-            {
-                return Task.FromException(releaseException);
-            }
-            else
-            {
-                return Task.FromException(scopeException!);
-            }
+            return ReleaseResourcesCore(scope);
```

Comment:
I refactored this code because it wasn't consistent depending on whether the path was synchronous or async or whether the values were being logged.